tests/test_fieldset.py

```diff
@@ -346,6 +349,88 @@ def test_vector_fields(mode, swapUV):
         assert abs(pset[0].lat - .5) < 1e-9
 
 
+@pytest.mark.parametrize('mode', ['scipy', 'jit'])
+@pytest.mark.parametrize('time_periodic', [4*86400.0, False])
+@pytest.mark.parametrize('cs', [False, 'auto', (1, 32, 32)])
+@pytest.mark.parametrize('with_GC', [False, True])
+def test_from_netcdf_memory_containment(mode, time_periodic, cs, with_GC):
+    class PerformanceLog():
+        samples = []
+        memory_steps = []
+        _iter = 0
+
+        def advance(self):
+            process = psutil.Process(os.getpid())
+            self.memory_steps.append(process.memory_info().rss)
+            self.samples.append(self._iter)
+            self._iter += 1
+
+    def perIterGC():
+        gc.collect()
+
+    def periodicBoundaryConditions(particle, fieldset, time):
+        while particle.lon > 180.:
+            particle.lon -= 360.
+        while particle.lon < -180.:
+            particle.lon += 360.
+        while particle.lat > 90.:
+            particle.lat -= 180.
+        while particle.lat < -90.:
+            particle.lat += 180.
+
+    process = psutil.Process(os.getpid())
+    mem_0 = process.memory_info().rss
+    fnameU = path.join(path.dirname(__file__), 'test_data', 'perlinfieldsU.nc')
+    fnameV = path.join(path.dirname(__file__), 'test_data', 'perlinfieldsV.nc')
+    ufiles = [fnameU, ] * 4
+    vfiles = [fnameV, ] * 4
+    timestamps = np.arange(0, 4, 1) * 86400.0
+    timestamps = np.expand_dims(timestamps, 1)
+    files = {'U': ufiles, 'V': vfiles}
+    variables = {'U': 'vozocrtx', 'V': 'vomecrty'}
+    dimensions = {'lon': 'nav_lon', 'lat': 'nav_lat'}
+
+    fieldset = FieldSet.from_netcdf(files, variables, dimensions, timestamps=timestamps, time_periodic=time_periodic, allow_time_extrapolation=True if time_periodic in [False, None] else False, field_chunksize=cs)
+    perflog = PerformanceLog()
+    postProcessFuncs = [perflog.advance, ]
+    if with_GC:
+        postProcessFuncs.append(perIterGC)
+    pset = ParticleSet(fieldset=fieldset, pclass=ptype[mode], lon=[0.5, ], lat=[0.5, ])
+    mem_0 = process.memory_info().rss
+    mem_exhausted = False
+    try:
+        pset.execute(pset.Kernel(AdvectionRK4)+periodicBoundaryConditions, dt=delta(hours=1), runtime=delta(days=7), postIterationCallbacks=postProcessFuncs, callbackdt=delta(hours=12))
+    except MemoryError:
+        mem_exhausted = True
+    mem_steps_np = np.array(perflog.memory_steps)
+    if with_GC:
+        assert np.allclose(mem_steps_np[7:], perflog.memory_steps[-1], rtol=0.01)
+    if cs is not False or with_GC:
+        assert np.alltrue((mem_steps_np-mem_0) < 4712832)   # represents 4 x [U|V] * sizeof(field data)
+    assert not mem_exhausted
+
+
+@pytest.mark.parametrize('mode', ['scipy', 'jit'])
+@pytest.mark.parametrize('time_periodic', [4*86400.0, False])
+@pytest.mark.parametrize('cs', [False, 'auto', {'x': 32, 'y': 32}, {'time_counter': 1, 'x': 32, 'y': 32}, (32, 32), (1, 32, 32)])
```

Comment:
Also test here for simply `32`? Because I heard from some users that they expected that they could put an integer for `field_chunksize`

Reply:
No, we don't facilitate THAT right now, cause it's ambiguous - which data dimension does this apply to ? All of them ? If can be done, if requested, but it's also not great in performance cause the used dimensions are anisotropic - meaning: either the laterial dimensions are much larger than the vertical, or vice versa (for cross-section/sinking studies).

If you wish that to happen, we can integrate it, though I'd rather prefer that working separately in an issue - this one is really just for testing and fixing what should already work.

Reply:
Well, then better to throw an error if `field_chunksize` is not a tuple